systemtest/src/test/java/io/strimzi/systemtest/operators/LeaderElectionIsolatedST.java

```diff
@@ -0,0 +1,129 @@
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.systemtest.resources.ResourceManager.kubeClient;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+
+/**
+ * Suite for testing Leader Election feature <br>
+ *
+ * The feature allows users to run Cluster operator in more than one replica <br>
+ *
+ * There will be always one leader, other replicas will stay in "standby" mode <br>
+ *
+ * The whole procedure of deploying CO with Leader Election enabled and many more is described in
+ *
+ * <a href="https://strimzi.io/docs/operators/in-development/configuring.html#assembly-using-multiple-cluster-operator-replicas-str">the documentation</a>
+ */
+
+@Tag(REGRESSION)
+@IsolatedSuite
+public class LeaderElectionIsolatedST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(LeaderElectionIsolatedST.class);
+
+    /*
+     These env variables are already inside packaging/../060-Deployment-strimzi-cluster-operator.yaml, but to be sure that
+     we are setting everything correctly and the Leader Election is really enabled, we are adding it to additional envs
+     */
+    final EnvVar leaderEnabledEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_ENABLED")
+        .withValue("true")
+        .build();
+
+    final EnvVar leaseNameEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_LEASE_NAME")
+        .withValue(Constants.STRIMZI_DEPLOYMENT_NAME)
+        .build();
+
+    final EnvVar leaseNamespaceEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_LEASE_NAMESPACE")
+        .withNewValueFrom()
+            .withNewFieldRef()
+                .withFieldPath("metadata.namespace")
+            .endFieldRef()
+        .endValueFrom()
+        .build();
+
+    final EnvVar leaderIdentityEnv = new EnvVarBuilder()
+        .withName("STRIMZI_LEADER_ELECTION_IDENTITY")
+        .withNewValueFrom()
+            .withNewFieldRef()
+                .withFieldPath("metadata.name")
+            .endFieldRef()
+        .endValueFrom()
+        .build();
+
+    final List<EnvVar> envList = Arrays.asList(leaderEnabledEnv, leaseNameEnv, leaseNamespaceEnv, leaderIdentityEnv);
+
+    @IsolatedTest
+    void testLeaderElection(ExtensionContext extensionContext) {
+        TestStorage testStorage = new TestStorage(extensionContext);
+
+        clusterOperator = clusterOperator.defaultInstallation()
+            .withExtensionContext(extensionContext)
+            .withExtraEnvVars(envList)
+            .withReplicas(2)
+            .createInstallation()
+            .runInstallation();
+
+        Lease oldLease = kubeClient().getClient().leases().inNamespace(testStorage.getNamespaceName()).withName(Constants.STRIMZI_DEPLOYMENT_NAME).get();
+        String oldLeaderPodName = oldLease.getSpec().getHolderIdentity();
```

Comment:
How do you know the operator is ready and has a leader elected at this point?

Reply:
We are waiting for the deployment readiness there. So I assume that when the deployment and pods are up and running, the leader is also elected.

Reply:
What exactly is _there_?

Reply:
On line 97 - `runInstallation()` - it deploys the CO and wait for readiness of `Deployment` and its pods

Reply:
Ok, fine.

But I think you guys really need to learn to name the test methods with what they actually do. The code would be much more readable for everyone if it was called for example `installAndWaitForReadiness`. This way, this is a hidden knowledge and makes it really hard to read and review the code because nobody without the knowledge knows whether this is a bug, an issue or what exactly happens there. Not saying you need to rename it in this PR, but this is not the first time I see it in the STs.

Reply:
I agree. For this I created - https://github.com/strimzi/strimzi-kafka-operator/issues/7368 . We will look into in one of the future PRs. Thanks @scholzj